### DISCUSSION TITLE
How to use private icons instead of public icons in controls that support icons?

### DISCUSSION THREAD

**Discussion by bao-cn** [lepoco/wpfui#366] [Q&A]
The desired icon cannot be found in the list of supported icons. How to make the control support private icons?

**Comment by pomianowski** [comment]
Hi @6get-xiaofan, you can add a new font like _Font Awesome_ and use it in `FontIcon`.

1. Add the font file in TTF format as the resource of your application.
```xml
<ItemGroup>
    <Resource Include="Fonts\FontAwesome.ttf" />
</ItemGroup>
```

2. Add the font as a resource for the controls.
```xml
<Application
    ...>
    <Application.Resources>
        <ResourceDictionary>
            <ResourceDictionary.MergedDictionaries>
                <ui:ThemesDictionary Theme="Dark" />
                <ui:ControlsDictionary />
                <FontFamily x:Key="FontAwesomeFont">pack://application:,,,/;component/Fonts/#Font Awesome</FontFamily>
            </ResourceDictionary.MergedDictionaries>
        </ResourceDictionary>
    </Application.Resources>
</Application>
```

3. In the `FontIcon`, indicate that you want to use this font.
```xml
<ui:FontIcon
    FontFamily="{DynamicResource FontAwesomeFont}"
    Glyph="&#xe00b;" />
```

**Reply by bao-cn** [reply]
![image](https://user-images.githubusercontent.com/57002549/195003909-e4396102-ab67-483a-bf65-9087c5288413.png)

As shown in the figure, I want to change the icons here to other icons. How to set them? Thank you.

![image](https://user-images.githubusercontent.com/57002549/195004105-5c463052-9c6e-4307-97e5-56993b566a58.png)

I made an icon ttf myself
